hamdy-farg/IBM_flutter_final_project
# Features :Adding feature add new Room -  update add new workspace .

_**- Designed and implemented a user-friendly UI form for adding new rooms.**_  👍 
![image](https://github.com/user-attachments/assets/8d245ab6-3810-4d57-889b-4cfbc8e540bd)
![image](https://github.com/user-attachments/assets/e61bca3d-648e-47a2-a7be-51a3e700646e)
Implemented a dedicated AddNewRoomCubit to handle user interactions, form data updates, and state management.
Defined various states in AddNewRoomState to reflect UI updates and ensure reactive feedback.
The form includes the following fields:
Title: Accepts a meaningful room title.
Description: A brief description of the room.
Capacity: Specifies the number of seats available.
Image Picker: Allows users to upload an image of the room.
Date Pickers: To specify the start and end dates for room availability.
Time Pickers: To specify the time range for room availability on selected dates.


_**- Implement Reusable Components and make it Global to use in another places**_

Widgets Added :
Custom Date Picker:
A reusable widget to select start and end dates.
![image](https://github.com/user-attachments/assets/8bfcaea6-8d14-42b1-8e05-a7aa84274912)

Includes validation logic to ensure the start date is earlier than the end date.
![image](https://github.com/user-attachments/assets/223d79e0-9e12-4088-a150-f18c69d92f89)



Custom Time Picker:
A reusable widget to select start and end times.
![image](https://github.com/user-attachments/assets/4fce12fe-76ff-44e7-bd9e-624130f19e82)

- !!!!! doesn't includes validation to ensure the end time is later than the start time. 
- ⚠️⚠️⚠️⚠️ 
![image](https://github.com/user-attachments/assets/5e8915da-2f29-4808-ad4e-b787fce06f60)


Logic 🧠 
AddNewRoomCubit:
Manages the state for all form inputs (e.g., title, description, capacity, dates, and times).
Includes methods for real-time validation and updates.
Stores all input data for future submission to the backend.


How to Test 🧪 
Navigate to the Add New Room Screen:
Access the screen from the navigation menu or another relevant entry point.
Fill Out the Form:

Enter valid and invalid inputs for each field to test validation.

![image](https://github.com/user-attachments/assets/0342f3ba-055f-427c-a043-c20f22148432)

Upload an image for the room.
Select valid and invalid date/time ranges to check validation logic.
Submit the Form:

Ensure the form is submitted successfully when all inputs are valid.
Check for appropriate error messages when validation fails.
